services/api/service.go

```diff
@@ -2228,6 +2269,80 @@ func (api *RelayAPI) handleSubmitNewBlock(w http.ResponseWriter, req *http.Reque
 	w.WriteHeader(http.StatusOK)
 }
 
+func (api *RelayAPI) saveBlockSubmissionMetrics(pf common.Profile) {
+	metrics.SubmitNewBlockCount.Add(
+		context.Background(),
+		1,
+		otelapi.WithAttributes(
+			attribute.String("contentType", pf.ContentType),
+			attribute.Bool("isGzip", pf.IsGzip),
+			attribute.Bool("aboveFloorBid", pf.AboveFloorBid),
+			attribute.Bool("simulationSuccess", pf.SimulationSuccess),
+			attribute.Bool("wasBidSaved", pf.WasBidSaved),
+			attribute.Bool("optimistic", pf.Optimistic),
+		),
```

Comment:
**issue:**

this will cause undesirable result (see another comment).

for example, the following code:

```go
metrics.BuilderDemotionCount.Add(context.Background(), 1)
metrics.BuilderDemotionCount.Add(context.Background(), 1, otelapi.WithAttributes(
	attribute.String("foo", "bar"),
))
metrics.BuilderDemotionCount.Add(context.Background(), 1, otelapi.WithAttributes(
	attribute.String("foo", "qux"),
))
```

.. causes the following metrics to be reported:

```text
mev_boost_relay_builder_demotion_count_total{otel_scope_name="mev-boost-relay",otel_scope_version=""} 1
mev_boost_relay_builder_demotion_count_total{foo="bar",otel_scope_name="mev-boost-relay",otel_scope_version=""} 1
mev_boost_relay_builder_demotion_count_total{foo="qux",otel_scope_name="mev-boost-relay",otel_scope_version=""} 1
```

Reply:
all the above attributes have only 2 possible values and would not cause the counter to grow infinitely. There's also the benefit of analysing the latency based on these attributes. Is that possible with regard to labels?

Reply:
64 permutations then => that's ok, I guess.